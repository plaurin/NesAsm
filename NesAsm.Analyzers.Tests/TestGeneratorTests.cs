using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp;

namespace NesAsm.Analyzers.Tests;

public class TestGeneratorTests
{
    [Fact]
    [UseReporter(typeof(VisualStudioReporter))]
    public void TestGenerator()
    {
        var generator = new TestGenerator();
        var source = @"
namespace MyCode
{
    public class Program
    {
        public static void Main(string[] args)
        {
        }
    }
}
";

        GeneratorTestUtilities.TestGenerator(generator, source);
    }

    [Fact]
    [UseReporter(typeof(VisualStudioReporter))]
    public void TestIncrementalGenerator()
    {
        var generator = new Generator();
        var source = (@"
using NesAsm.Emulator;
namespace MyCode;
public class Program : NesAsm.Emulator.ScriptBase
{
    public Program(NESEmulator emulator) : base(emulator) {}
}
");

        GeneratorTestUtilities.TestGenerator(generator, source);
    }
}