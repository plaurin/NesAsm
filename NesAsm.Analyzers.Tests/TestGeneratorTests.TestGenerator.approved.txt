- Main diagnostics:
- SyntaxTrees Count: 2
- Output diagnostics:
NesAsm.Analyzers\NesAsm.Analyzers.TestGenerator\test.g.cs(1,1): hidden CS8019: Unnecessary using directive.
- RunResult.GeneratedTrees Count: 1
- RunResults diagnostics:
- RunResult[0].Generator: NesAsm.Analyzers.TestGenerator
- RunResult[0] diagnostics:
- RunResult[0].GeneratedSources Count: 1
- RunResult[0].Exception: 
- RunResult[0].GeneratedSources[0].HintName: test.g.cs
- RunResult[0].GeneratedSources[0].SourceText:
using System;
